src/libsystemd-network/network-common.c

```diff
@@ -13,3 +14,90 @@ int get_ifname(int ifindex, char **ifname) {
 
         return format_ifname_alloc(ifindex, ifname);
 }
+
+usec_t unaligned_be32_sec_to_usec(const void *p, bool max_as_infinity) {
+        uint32_t s = unaligned_read_be32(ASSERT_PTR(p));
+
+        if (s == UINT32_MAX && max_as_infinity)
+                return USEC_INFINITY;
+
+        return s * USEC_PER_SEC;
+}
+
+usec_t be32_sec_to_usec(be32_t t, bool max_as_infinity) {
+        uint32_t s = be32toh(t);
+
+        if (s == UINT32_MAX && max_as_infinity)
+                return USEC_INFINITY;
+
+        return s * USEC_PER_SEC;
+}
+
+usec_t be32_msec_to_usec(be32_t t, bool max_as_infinity) {
+        uint32_t s = be32toh(t);
+
+        if (s == UINT32_MAX && max_as_infinity)
+                return USEC_INFINITY;
+
+        return s * USEC_PER_MSEC;
+}
+
+usec_t be16_sec_to_usec(be16_t t, bool max_as_infinity) {
+        uint16_t s = be16toh(t);
+
+        if (s == UINT16_MAX && max_as_infinity)
+                return USEC_INFINITY;
+
+        return s * USEC_PER_SEC;
+}
+
+be32_t usec_to_be32_sec(usec_t t) {
+        if (t == USEC_INFINITY)
+                /* UINT32_MAX may be handled as infinity. */
+                return htobe32(UINT32_MAX);
+
+        assert(t <= UINT32_MAX * USEC_PER_SEC);
+        return htobe32((uint32_t) DIV_ROUND_UP(t, USEC_PER_SEC));
+}
+
+be32_t usec_to_be32_sec_safe(usec_t t) {
+        /* This is similar to usec_to_be32_sec(), but this handles too large values gracefully. */
```

Comment:
ah, right, here you have the saturated version anyway. maybe then just use the saturated version everyhwere? simpler, safer, more obvious? drop the one with the assert?

(and maybe rename this function to usec_to_be32_sec_saturate())?

Reply:
Done. But `_saturate()` is too verbose. The functions are (currently) only used in libsystemd-network, hence I'd like to keep the name as is.